### ISSUE TITLE
Feature: Add sync capabilities

### ISSUE THREAD

**Issue by titoesteves** [BoostIO/Boostnote#308]
A crucial feature for any notes app is the ability to sync with multiple devices. It would be great if this app would be able to sync with simplenote or evernote as they both seem to support markdown.

**Comment by asmsuechan** [comment]
Hi @titoesteves, I've heard it from many people. Then I always answer like this: I don't wanna prepare for sync (e.g. user account, encrypted data...etc). However now I'm contemplating about GitHub integration for Boostnote. I'll open an issue about it. Please comment anything you think if you're interested.

**Comment by titoesteves** [comment]
@asmsuechan I never thought about GitHub integration. That would be quite interesting actually. I guess the one downside is that you would not be able to utilize the GitHub mobile app as note app client. Whereas with simplenote/evernote you can use the mobile app versus as a note app client.

On another note, I tried out Boostnote today and I have to say it is really great! Much nicer than both evernote and simplenote. Add sync capabilities and it is the perfect note taking app for developers.

**Comment by twhiting** [comment]
I've mentioned a github integration to the previous maintainer, it'd be great.  @titoesteves I store my boostnote data in a Google Drive folder which syncs to multiple devices with no problems. Just thought I'd let you know for short term.  Onedrive works great too.

**Comment by Aivean** [comment]
I think Evernote sync is a must-have. A lot of developers (including me) are using Evernote day-to-day (especially on mobile devices) and some alternative software with markdown support for complex notes.

For example, I use [Marxi.co](http://marxi.co/). It's good, but not perfect (paid, buggy, unmaintained). However, it's integration with Evernote is worth mentioning, as, I think, it's as good as it could be.

To conclude, I think that adding Evernote support is the way to attract new users. People who use evernote already wont switch to an alternative (unless alternative is as cross-platform and feature-rich). However, they will use an alternative **in addition** to evernote to get markdown support.

**Comment by asmsuechan** [comment]
Hi, @Aivean. Thanks for your feedback. Have you ever tried [ever2boost](https://github.com/BoostIO/ever2boost)? I assume the data migration or sync from Evernote on local is complicated. But it you say is absolutely true, so I'll try it.

**Comment by Aivean** [comment]
Hi @asmsuechan. Thank you for your work. 

I checked `ever2boost`, but I didn't try it. My point is that people that use Evernote usually don't want to switch over from it. Evernote is a great project (perhaps a leader in the field), it's cross-platform and very feature-rich.  I'm mostly happy with Evernote, the only feature that it lacks, in my opinion, is markdown support.

I believe that your project can attract users like me if it has integration with evernote. Users that will use it **together with** evernote, **not as a replacement of** evernote.

**Comment by iliyang** [comment]
Syncing could be done by simply storing your directory on a cloud drive, as @twhiting suggests. The issue is how to then access the notes on your mobile device. There are many plain-text mobile editors that supprt Markdown, but unfortunately Boostnote's format, while simple `.cson`, is not exactly plain text. So it's incompatible with those apps. Now, if notes would be stored as basic markdown files, with the metadata of all notes in a storage kept in a single separate metadata file, then you have your cross-platform sync!

**Comment by titoesteves** [comment]
I agree. The primary goal of having sync IMO is so that you can access notes on your mobile device. The ever2boost cli tool seems to be a migration tool but does not address the ability to access notes on mobile devices. Storing directory on a cloud drive also does not achieve this. Seems there is already a feature request for mobile clients already: https://github.com/BoostIO/Boostnote/issues/335.

**Comment by GaryFurash** [comment]
Don't use github integration. The whole point of your tool is to avoid github, where you're private gists aren't really private.

**Comment by twhiting** [comment]
@GaryFurash the idea would be to use a private repository to sync your data using github.  Not open gists.

**Comment by kocoai** [comment]
+1 for sync, and github gist is more dev friendly than box, dropbox, drive & co... in my opinion ^^

**Comment by residentbrit** [comment]
Hi, any sync would be welcome, I am trialing Boostnote on my Mac and my Ubuntu laptops using Dropbox as my storage. After installing Boostnote on the Mac and then the Linux computer, and pointing to my Dropbox mounted share I ended up with duplicate initial notes which is no biggie, but any changes made on one computer's Boostnote are not seen by the other computer's Boostnote until the app is restarted.   Could you at least have the app periodically check the "my storage" folder for changes?  

I do use Evernote also but I want to partition some of my work outside of Evernote, I've tried, Tagspaces and numerous other note sharing apps, but I'd really like Boostnote to be the one to fill the gap for me.

**Comment by ynCode** [comment]
+1, but in china, you know, we can't use great service such as google、github、Dropbox ……a better way is supporting customize repository，thx！

**Comment by hijak** [comment]
i use minio's mc client to sync my boost notes folder this in combination with a launchd script I'm happy :)

`mc mirror --force --remove --watch /LOCAL/PATH MINIO/BUCKET/`

Generate your launchd script here: http://launched.zerowidth.com

**Comment by tiagoboeing** [comment]
I was writing about this. Please, sync capabilities is important.
I use in home and work, in two computers is hard to have different notes in each one and not have a tool for sync.

**Comment by BobTB** [comment]
Best would be to use the standardfile for sync, completely private, can be self hosted, and its encrypted end to end, works great, and if boostnote could use it, would be really great
https://standardfile.org/
